contracts/sources/obligation.move

```diff
@@ -693,47 +694,103 @@ module suilend::obligation {
 
     // === Private Functions ===
     fun is_looped<P>(obligation: &Obligation<P>): bool {
-        let usdc_reserve_array_index = 1;
-        let usdt_reserve_array_index = 2;
+        let target_reserve_array_indices = vector[
+            1, 2, 5
+        ];
+
+        // The vector target_reserve_array_indices maps to disabled_pairings_map
+        // by corresponding indices of each element
+        // target_reserve_index --> pairings disabled
+        let disabled_pairings_map = vector[
+            vector[2, 5], // 1 --> [2, 5]
+            vector[1, 5], // 2 --> [1, 5]
+            vector[1, 2], // 5 --> [1, 2]
+        ];
 
         let i = 0;
         while (i < vector::length(&obligation.borrows)) {
             let borrow = vector::borrow(&obligation.borrows, i);
 
-            let deposit_index = find_deposit_index_by_reserve_array_index(
-                obligation, 
-                borrow.reserve_array_index
+            let is_borrow_looped = is_borrow_looped(
+                obligation,
+                borrow,
+                &target_reserve_array_indices,
+                &disabled_pairings_map,
             );
 
-            if (deposit_index < vector::length(&obligation.deposits)) {
+            if (is_borrow_looped) {
                 return true
             };
 
-            // special case for usdc/usdt looping
-            if (borrow.reserve_array_index == usdc_reserve_array_index) {
-                let usdt_deposit_index = find_deposit_index_by_reserve_array_index(
-                    obligation, 
-                    usdt_reserve_array_index
-                );
-                if (usdt_deposit_index < vector::length(&obligation.deposits)) {
-                    return true
-                };
-            };
+            i = i + 1;
+        };
 
-            if (borrow.reserve_array_index == usdt_reserve_array_index) {
-                let usdc_deposit_index = find_deposit_index_by_reserve_array_index(
+        false
+    }
+    
+    // Checks if a given `Borrow` object is looped
+    fun is_borrow_looped<P>(
```

Comment:
nit: don't abstract this. i think it'll keep the diff smaller which is easier for me to follow

Reply:
done